EIPS/eip-9174.md

```diff
@@ -0,0 +1,144 @@
+---
+eip: 9174
+title: On-chain upgrade signaling
+description: Allows participants to indicate readiness for a client upgrade when producing blocks
+author: William Entriken (@fulldecent)
+discussions-to: https://ethereum-magicians.org/t/eip-xxxx-on-chain-upgrade-signaling/22306
```

Comment:
Please use the following template for the Eth Magicians discussion rather than copying/pasting the EIP contents
------------

t is recommended to use the following template when creating a `discussion-to` thread for your EIP on ethereum-magicians.org. Each section has instructions set within a blockquote. Please read the instructions carefully before posting the topic. 

**Delete the blockquoted instructions below and everything above the following emdash line before posting your topic.**
 
--

Discussion topic for EIP-XXXX <link to EIP>

#### Update Log 

> This section should list significant updates to the EIP as the specification evolves. The first entry should be the PR to create the EIP. The recommended format for log entries is:
> * yyyy-mm-dd: Single sentence description, commit [link to commit]()
> 
> For example, using EIP-1: 
> * 2024-06-05: Enable external links to Chain Agnostic Improvement Proposals (CAIPs), commit [32dc740](https://github.com/ethereum/EIPs/commit/32dc74013133b579a76f27ea50a219852ed0fa71).
> ...
>
> Fill in the log for the EIP's initial draft below:

* yyyy-mm-dd: initial draft <link to pull request>

#### External Reviews

> This section should list notable reviews the EIP has received from the Ethereum community. These can include specific comments on this forum, timestamped audio/video exchanges, formal audits, or other external resources. This section should be the go-to for readers to understand the community's current assessment of the EIP. Aim for neutrality, quality & thoroughness over "cherry-picking" the most favorable reviews. 
> 
> The recommended format for entries is: 
> 
> * yyyy-mm-dd: Single sentence description, [link to review]()
> 
> For example, using EIP-1559, one entry could be:
> * 2020-12-01: "An Economic Analysis of EIP-1559", by Tim Roughgarden, [full report](https://timroughgarden.org/papers/eip1559.pdf)

None as of yyyy-mm-dd.

#### Outstanding Issues 

> This section should highlight outstanding issues about the EIP, and, if possible, link to forums where these are being addressed. This section should allow readers to quickly understand what the most important TODOs for the EIP are, and how to best contribute. Once issues are resolved, they should be checked off with a note giving context on the resolution. 
> 
> The recommended format for new entries is: 
> * [ ] yyyy-mm-dd: Issue description, [link to issue]()
>
> Once issues are addressed, these becomes: 
> * [x] yyyy-mm-dd: Issue description, [link to issue]()
>     * yyyy-mm-dd: Resolution description, [link to resolution]()
> 
> For example, using EIP-3675, one entry could be: 
> * [x] 2021-07-08: Repurpose the `DIFFICULTY` opcode, [tracking issue](https://github.com/ethereum/pm/commit/496b4dc9bcb76e5290f5cc5d52d7424ae75a4f96)
>     * 2021-10-30: Introduce EIP-4399, [EIP PR](https://github.com/ethereum/EIPs/pull/4399)
>
> If there are any known outstanding issues currently, fill them out now.

None as of yyyy-mm-dd.

Reply:
Thank you, added at https://ethereum-magicians.org/t/eip-7848-on-chain-upgrade-signaling/22306